src/components/register.tsx

```diff
@@ -0,0 +1,147 @@
+import { Input } from "./ui/input";
+import { Button } from "./ui/button";
+import { Card, CardHeader, CardTitle, CardContent, CardDescription } from "./ui/card";
+import * as Yup from 'yup';
+import { useFormik } from 'formik';
+import React, { useEffect, useState } from 'react';
+import { signIn } from "next-auth/react";
+import useAxiosAuth from "@/lib/hooks/useAxiosAuth";
+
+const validationSchema = Yup.object({
+  email: Yup.string()
+    .email('O campo de email deve ser um e-mail válido')
+    .max(200, 'O campo de email deve ter menos de 200 caracteres')
+    .required('O campo de email deve ser preenchido')
+    .trim()
+    .lowercase(),
+  name: Yup.string()
+    .max(200, 'O campo de nome deve ter menos de 200 caracteres')
+    .required('O campo de nome deve ser preenchido')
+    .trim(),
+  password: Yup.string()
+    .min(3, 'O campo password deve ter mais de 3 caracteres')
+    .max(255, 'O campo password deve ter menos de 255 caracteres')
+    .matches(/^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*\W|_)[\S]{8,}$/, {
+      message: 'A senha deve conter no mínimo 8 caracteres, uma letra maiúscula, uma letra minúscula, um número e um caractere especial',
+    })
+    .required('O campo de senha é obrigatório')
+    .trim(),
+});
+
+type TRegisterComponentProps = {}
+
+const RegisterComponent = ({ }: TRegisterComponentProps) => {
+  const [messageErrorRegister, setMessageErrorRegister] = useState<string | null>(null);
+  const [inRequest, setInRequest] = useState<boolean>(false);
+
+  useEffect(() => {
+    return () => {
+      setMessageErrorRegister(null);
+    }
+  }, []);
+
+  const authAxios = useAxiosAuth();
+
+  async function registerUser(data: {
+    email: string;
+    name: string;
+    password: string;
+  }) {
+    return await authAxios.post('/api/auth/register', data)
+  }
+
+  const formik = useFormik({
+    initialValues: {
+      email: '',
+      name: '',
+      password: '',
+    },
+    validationSchema,
+    onSubmit: (values) => {
+      handleRegister(values);
+    },
+  });
+
+  const handleRegister = async (values: {
+    email: string;
+    name: string;
+    password: string;
+  }) => {
+    try {
+      setInRequest(true);
+      await registerUser(values);
+      await signIn('credentials', {
+        redirect: true,
+        email: values.email,
+        password: values.password,
+      });
+    } catch (error) {
+      console.error(error);
+      setMessageErrorRegister("Erro ao registrar o usuário, valide os dados preenchidos");
```

Comment:
Notei que na hora de registrar, pode acontecer de mesmo o registro sendo feito com sucesso (usuário adicionado à base), essa mensagem pode aparecer. Apesar disso, se não me engano isso foi uma reflexão do server também, certo? Onde o cadastro ocorria mas poderia se receber um erro por algum detalhe da verificação por e-mail? Se for o caso, já que está funcionando, poderiamos deixar a mensagem sempre como uma de sucesso, por enquanto.

Reply:
Faz sentido